tests/integration_python/test_edge_cases.py

```diff
@@ -470,6 +470,362 @@ def test_pypi_url_fragment_in_project_deps(tmp_pixi_workspace: Path, pixi: Path)
         pytest.fail("Failed to solve the pypi requirements. pytrace=False")
 
 
+def test_pypi_external_source_incorrect_hash(tmp_pixi_workspace: Path, pixi: Path) -> None:
+    """Test installing an external source with pyproject.toml with an incorrect hash"""
+    pyproject_content = """
+[project]
+version = "0.1.0"
+name = "test"
+requires-python = "== 3.12"
+dependencies = [
+    "requests @ https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl#sha256=8d29f97ed1541709b57caddb77bb20592411d7ca10ec4f03275f49ee8456e230"
+]
+[tool.pixi.workspace]
+platforms = ["linux-64", "osx-arm64", "win-64"]
+channels = ["https://prefix.dev/conda-forge"]
+"""
+    pyproject_path = tmp_pixi_workspace / "pyproject.toml"
+    pyproject_path.write_text(pyproject_content)
+
+    src_dir = tmp_pixi_workspace / "src" / "test"
+    src_dir.mkdir(parents=True, exist_ok=True)
+    (src_dir / "__init__.py").touch()
+
+    # use verify_cli_command to check that the installation fails with the expected error
+    verify_cli_command(
+        [pixi, "install", "-v"],
+        cwd=tmp_pixi_workspace,
+        expected_exit_code=ExitCode.FAILURE,
+        stderr_contains=["Hash verification failed"],
+    )
+
+
+def test_pypi_external_source_correct_hash(tmp_pixi_workspace: Path, pixi: Path) -> None:
+    """Test installing an external source with correct hash succeeds"""
+    pyproject_content = """
+[project]
+version = "0.1.0"
+name = "test"
+requires-python = "== 3.12"
+dependencies = [
+    "requests @ https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl#sha256=70761cfe03c773ceb22aa2f671b4757976145175cdfca038c02654d061d6dcc6"
+]
+[tool.pixi.workspace]
+platforms = ["linux-64", "osx-arm64", "win-64"]
```

Comment:
TO reduce the overhead of testing we could include only the current platform here.